examples/dynamo/eg3/solver_mod.x90

```diff
@@ -437,7 +437,7 @@ contains
   call invoke_set_field_scalar(1.0_r_def, Ax)
   call invoke( matrix_vector_kernel_mm_type(lumped_weight,Ax,mm) )
 
-  !PSY call invoke ( divide_field(rhs, lumped_weight, lhs))
+  !PSY call invoke ( divide_fields(rhs, lumped_weight, lhs))
```

Comment:
OK, I'm happy to defer, thanks for trying. The lack of support for expressions is a known issue, not allowing the same argument more than once is a purposeful check as aliasing is not allowed in fortran and parallel directives might rely on there being no aliasing. I was not aware about errors with loop indices so I think that needs looking at. I have created two issues, #40 for updating solver_mod.x90 and #41 for fixing the error with loop indices.